src/index.js

```diff
@@ -72,15 +149,52 @@ class ClientAuthOIDC {
   }
 
   /**
-   * @return {Promise<string>} Resolves to current user's WebID URI
+   * Returns the previously selected provider (cached on the object, saved in
+   * localStorage, or loaded from the callback uri using a previously stored
+   * state parameter).
+   *
+   * @return {string|null}
+   */
+  currentProvider () {
+    return this.providerUri ||
+      this.store.getItem(CURRENT_PROVIDER) ||
+      this.providerFromCurrentUri()
+  }
+
+  /**
+   * Saves the currently selected provider in storage.
+   *
+   * @param providerUri {string}
+   */
+  saveCurrentProvider (providerUri) {
+    this.providerUri = providerUri
+
+    this.store.setItem(CURRENT_PROVIDER, providerUri)
+  }
+
+  /**
+   * Resolves with the currently logged in user's WebID URI.
+   * Recommended to call this as soon as the page is loaded (or your framework
+   * ready event fires).
+   *
+   * Attempts to load the logged in webid from storage, or from the callback
+   * redirect uri (this is the part that requires an async operation).
+   *
+   * @return {Promise<string>} WebID URI
    */
   currentUser () {
     if (this.webId) {
       return Promise.resolve(this.webId)
     }
 
-    // Attempt to find a provider based on the 'state' param of the current URI
-    let providerUri = this.providerFromCurrentUri()
+    this.loadCurrentCredentials()
+    if (this.webId) {
+      return Promise.resolve(this.webId)
+    }
```

Comment:
I was debating on what the best way to do this was. The idea is - it first checks `this.webId`, and only then tries to `loadCurrentCredentials()` (which potentially initializes `this.webId`) and retries checking it. I suppose it doesn't hurt to drop the first check, and just to always try and load the current credentials from storage. I'll change it.